common/authorization/interceptor.go

```diff
@@ -45,42 +50,79 @@ func (a *interceptor) Interceptor(
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
 
-	var namespace string
-	requestWithNamespace, ok := req.(requestWithNamespace)
-	if ok {
-		namespace = requestWithNamespace.GetNamespace()
-	}
+	var caller *Claims
```

Comment:
[Nit] interesting variable name. Why not just call it claims?

Reply:
ok I see we would shadow the variable in a different scope below. I'd suggest keeping this one named claims and change the one down below

Reply:
That's exactly what I did.